lib/spack/spack/cmd/env.py

```diff
@@ -391,6 +395,77 @@ def env_loads(args):
     print('   source %s' % loads_file)
 
 
+def env_update_setup_parser(subparser):
+    subparser.add_argument(
+        metavar='env', dest='env',
+        help='name or directory of the environment to activate'
+    )
+    spack.cmd.common.arguments.add_common_arguments(subparser, ['yes_to_all'])
+
+
+def env_update(args):
+    manifest_file = ev.manifest_file(args.env)
+    backup_file = manifest_file + ".bkp"
+    needs_update = not ev.is_latest_format(manifest_file)
+
+    if not needs_update:
+        tty.msg('No update needed for the environment "{0}"'.format(args.env))
+        return
+
+    proceed = True
+    if not args.yes_to_all:
+        msg = ('The environment "{0}" is going to be updated to the latest '
+               'schema format. This operation is not forward-compatible i.e.'
+               'old versions of Spack will not be able to use this environment'
+               ' anymore.')
+        tty.msg(msg.format(args.env))
+        proceed = tty.get_yes_or_no('Do you want to proceed?', default=False)
+
+    if not proceed:
+        tty.die('Operation aborted.')
+
+    ev.update_yaml(manifest_file, backup_file=backup_file)
+    msg = 'Environment "{0}" has been updated [backup={1}]'
+    tty.msg(msg.format(args.env, backup_file))
+
+
+def env_revert_setup_parser(subparser):
+    subparser.add_argument(
+        metavar='env', dest='env',
+        help='name or directory of the environment to activate'
+    )
+    spack.cmd.common.arguments.add_common_arguments(subparser, ['yes_to_all'])
+
+
+def env_revert(args):
+    manifest_file = ev.manifest_file(args.env)
+    backup_file = manifest_file + ".bkp"
+
+    # Check that both the spack.yaml and the backup exist, the inform user
+    # on what is going to happen and ask for confirmation
+    if not os.path.exists(manifest_file):
+        msg = 'cannot fine the manifest file of the environment [file={0}]'
+        tty.die(msg.format(manifest_file))
+    if not os.path.exists(backup_file):
+        msg = 'cannot find the old manifest file to be restored [file={0}]'
+        tty.die(msg.format(backup_file))
+
+    proceed = True
+    if not args.yes_to_all:
```

Comment:
It's true that these commands are called by the user to update config files/environments, but the prompt prints out relevant information on the scopes being updated and the fact that the new format is not forward-compatible:
```console
$ spack config update packages
==> Warning: the attribute "paths" in the "packages" section of the configuration has been deprecated [entry=CommentedMap([('cmake@3.17.3', '/usr')])]
==> The following configuration files are going to be updated to the latest schema format:

        [scope=user, file=/home/culpo/.spack/packages.yaml]

This operation is not forward-compatible.
==> Do you want to proceed? [y/N] 
```
Also I added the prompt since it's very likely that these commands are called once every so often from regular users, so interaction was not a major problem in my view. Let me know if you think it's still worth removing the prompt after this extra explanation and I'll proceed with that.

Reply:
Since users will end up calling this once for all scopes together, I think this won't actually be too tedious. I like the message.